### ISSUE TITLE
How to transform chart type

### ISSUE THREAD

**Issue by bkCDL** [Glavin001/ember-c3#6]
First of all, this is a superb project. It is exactly what I've been looking for in my reporting app.

I'm wondering how I can transform a pie graph into a bar graph. I see that with C3, one calls `chart.transform('bar')`. Is there a way right now to do this in Ember-C3? I looked at the code but didn't see anything obvious.

I tried this in my Ember template, but it ends up showing two charts on top of or interlaced with each other after I change the controller properties cited below:

```
{{#if isChartTypeBar}}
  {{c3-chart data=contentChart legend=chartLegendSettings tooltip=chartTooltipFormat axis=chartCatAxisBar}}
{{/if}}
{{#if isChartTypePie}}
  {{c3-chart data=contentChart legend=chartLegendSettings tooltip=chartTooltipFormat pie=chartPieFormat}}
{{/if}}
```

Then I realized I could clean up my code, and I changed my template to have merely a single `{{c3-chart ...}}` object with all the proper component properties for every possible chart type, like this:

```
{{c3-chart
  data=contentChart
  axis=chartAxis
  bar=chartBarFormat
  pie=chartPieFormat
  legend=chartLegendSettings
  tooltip=chartTooltipFormat
}}
```

When I changed the "type" value within the "data" property in my controller. It still gave me a multiple charts piled on top of each other.

Any thoughts on this?

Bryan

**Comment by JKGisMe** [comment]
> Not as of yet. We currently only update the graph when `data` changes 

Erm... but `type` is in `data`...
```
data = {
  rows: //arrays or whatever,
  type: 'area',
  colors: //define the colors
  hide: ['data1', 'data2']
}
```
Then, I have an observer that does `this.set('data.hide', ['data2', 'data3'])`

I have been setting all that to `data` for the component. I've been trying to change which info I hide in the graph with a user select because I don't want to iterate through all my rows separating out the elements to feed them into the chart individually but it doesn't seem to respond to the changes when i change the `hide` property. I will try Bryan's way and see if that works.

**Comment by maxwondercorn** [comment]
pr #79 exposes c3 chart api, so you can dynamically change the chart, load, unload data etc. Updated add-on will be published after checklist in issue #77 are completed
